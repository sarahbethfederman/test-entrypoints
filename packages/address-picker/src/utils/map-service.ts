export default class MapService {
  private placesService: google.maps.places.PlacesService = {} as google.maps.places.PlacesService;
  private autocompleteService: google.maps.places.AutocompleteService = new google.maps.places.AutocompleteService();
  private sessionToken: google.maps.places.AutocompleteSessionToken = new google.maps.places.AutocompleteSessionToken();

  constructor(map?: google.maps.Map) {
    this.placesService = new google.maps.places.PlacesService(map || document.createElement('div'));
  }

  /**
   * Map enum to string
   * @param status google.maps.places.PlacesServiceStatus
   */
  private apiRejectionReason(status: google.maps.places.PlacesServiceStatus): Error {
    if (status === google.maps.places.PlacesServiceStatus.ZERO_RESULTS) {
      return new Error('ZERO_RESULTS');
    } else if (status === google.maps.places.PlacesServiceStatus.INVALID_REQUEST) {
      return new Error('INVALID_REQUEST');
    } else if (status === google.maps.places.PlacesServiceStatus.OVER_QUERY_LIMIT) {
      return new Error('OVER_QUERY_LIMIT');
    } else if (status === google.maps.places.PlacesServiceStatus.NOT_FOUND) {
      return new Error('NOT_FOUND');
    } else if (status === google.maps.places.PlacesServiceStatus.REQUEST_DENIED) {
      return new Error('REQUEST_DENIED');
    } else if (status === google.maps.places.PlacesServiceStatus.UNKNOWN_ERROR) {
      return new Error('UNKNOWN_ERROR');
    }
    return new Error('UNKNOWN_ERROR');
  }

  public placePredictions(
    input: string,
    componentRestrictions: google.maps.places.ComponentRestrictions
  ): Promise<google.maps.places.AutocompletePrediction[]> {
    return new Promise((resolve, reject) => {
      (this.autocompleteService as google.maps.places.AutocompleteService).getPlacePredictions(
        {
          input,
          sessionToken: this.sessionToken,
          componentRestrictions,
        },
        (suggestions: google.maps.places.AutocompletePrediction[], status: google.maps.places.PlacesServiceStatus) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && suggestions !== null) {
            resolve(suggestions);
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }

  public placeDetails(
    placeId: string
  ): Promise<{
    geometry: google.maps.places.PlaceGeometry | undefined;
    addressComponents: google.maps.GeocoderAddressComponent[];
  }> {
    return new Promise((resolve, reject) => {
      (this.placesService as google.maps.places.PlacesService).getDetails(
        {
          placeId,
          sessionToken: this.sessionToken,
        },
        (
          { geometry, address_components = [] }: google.maps.places.PlaceResult,
          status: google.maps.places.PlacesServiceStatus
        ) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && address_components.length > 0) {
            resolve({
              geometry,
              addressComponents: address_components,
            });
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }
}
