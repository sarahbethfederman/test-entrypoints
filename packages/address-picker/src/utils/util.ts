import { AddressObject } from '../components/Modal';
import { STREET_TYPE } from '../components/Modal/constants';
import * as to from 'to-case';

/**
 * Where possible this function takes the users input and
 * Google's address information and enhances it with more
 * accurate subpremise/unit information
 */
export const transformGoogleResponse = (label: string, res: google.maps.GeocoderAddressComponent[]): AddressObject => {
  label = label.toLowerCase();

  // pick the easily determined properties from response
  const { short_name: postcode = '' } = res.find((curr) => curr.types.includes('postal_code')) || {};
  let { short_name: state = '' } = res.find((curr) => curr.types.includes('administrative_area_level_1')) || {};
  let { long_name: streetName = '' } = res.find((curr) => curr.types.includes('route')) || {};
  let { short_name: streetNumber = '' } = res.find((curr) => curr.types.includes('street_number')) || {};
  const { long_name: suburb = '' } = res.find((curr) => curr.types.includes('locality')) || {};
  const { long_name: country = '' } = res.find((curr) => curr.types.includes('country')) || {};
  let unit = '';

  // Google has no concept of "street". See if we recognise a streetType from
  // the route then use that to generate streetName and streetType
  const streetType = String(
    STREET_TYPE.reduce((acc, { value = '' }) => {
      if (streetName.includes(value)) {
        return acc + value.toLowerCase();
      }
      return acc;
    }, '')
  );
  streetName = streetName
    .toLowerCase()
    .replace(streetType, '')
    .trim();

  // edge case, state is listed as JBT for jervis bay when it should be ACT. Naughty Google
  if (state === 'JBT') {
    state = 'ACT';
  }

  // check for unit / subpremise info by crossreferencing against user input
  const streetNumberData = label.substring(0, label.indexOf(streetName)).trim();
  if (streetNumberData !== streetNumber) {
    if (streetNumberData.includes('/')) {
      unit = streetNumberData.substring(0, streetNumberData.indexOf('/'));
    } else {
      streetNumber = streetNumberData;
    }
  }

  return {
    unit,
    lotSection: '',
    level: '',
    buildingName: '',
    streetNumber,
    streetName,
    streetType,
    suburb: suburb.toLowerCase(),
    postcode,
    state,
    country: country.toLowerCase(),
  };
};

export const getFormatedString = (address: AddressObject) => {
  let formatString = address.unit ? ''.concat(address.unit, '/') : '';
  formatString = address.lotSection ? formatString.concat('Lot. ', address.lotSection) : formatString;
  formatString = address.level ? formatString.concat(' level ', address.level) : formatString;
  formatString = address.buildingName ? formatString.concat(' ', to.pascal(address.buildingName), ' ') : formatString;
  formatString = address.streetNumber ? formatString.concat(address.streetNumber) : formatString;
  formatString = address.streetName ? formatString.concat(' ', to.pascal(address.streetName)) : formatString;
  formatString = address.streetType ? formatString.concat(' ', to.pascal(address.streetType)) : formatString;
  formatString = address.suburb ? formatString.concat(', ', to.pascal(address.suburb)) : formatString;
  formatString = address.state ? formatString.concat(', ', to.upper(address.state)) : formatString;
  formatString = address.postcode ? formatString.concat(' ', address.postcode) : formatString;
  formatString = address.country ? formatString.concat(', ', to.pascal(address.country)) : formatString;
  return formatString;
};
