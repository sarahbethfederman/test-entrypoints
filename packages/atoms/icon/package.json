{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
