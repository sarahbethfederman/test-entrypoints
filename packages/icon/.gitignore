src/icons-compiled/*
src/index.ts