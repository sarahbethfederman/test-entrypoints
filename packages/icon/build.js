const SVGSpriter = require('svg-sprite');
const fs = require('fs');
const path = require('path');
const mkdirp = require('mkdirp');

const directory = `${__dirname}/src/icons`;

const spritify = new SVGSpriter({
  dest: 'dist',
  mode: {
    symbol: true,
    inline: true,
  },
});

fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(err);
    return;
  }

  files.forEach((file) => {
    if (file.endsWith('.svg')) {
      spritify.add(
        path.resolve(`${directory}/${file}`),
        file,
        fs.readFileSync(`${directory}/${file}`, { encoding: 'utf-8' })
      );
    }
  });

  spritify.compile(function(error, result, data) {
    for (var mode in result) {
      for (var resource in result[mode]) {
        fs.writeFileSync(`../../website/static/sprite.symbol.svg`, result[mode][resource].contents);
      }
    }
  });
});
