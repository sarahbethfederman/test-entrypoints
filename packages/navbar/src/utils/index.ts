export const browserSupportsPassiveListeners = () => {
  let supportsPassive = false;
  try {
    var opts = Object.defineProperty({}, 'passive', {
      get: function() {
        supportsPassive = true;
      },
    });
    // for testing browser compatibility, no need to worry
    // about the null TS issue.
    // @ts-ignore
    window.addEventListener('testPassive', null, opts);
    // @ts-ignore
    window.removeEventListener('testPassive', null, opts);
  } catch (e) {}

  return supportsPassive;
};
