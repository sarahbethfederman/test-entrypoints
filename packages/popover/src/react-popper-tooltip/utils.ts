import * as React from 'react';

export const TooltipContext = React.createContext({});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Fn = (...args: any[]) => void;
type FnOrBool = undefined | Fn;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const callAll = (...fns: FnOrBool[]) => (...args: any[]) => fns.forEach((fn) => fn && fn(...args));

export const noop = () => {};

export const canUseDOM = () => !!(typeof window !== 'undefined' && window.document && window.document.createElement);
