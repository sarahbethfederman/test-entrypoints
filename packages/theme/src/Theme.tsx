import * as React from 'react';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;
  }

  input::-ms-clear {
    display: none;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors['shade.700']};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>
        {children}
        <GlobalStyle />
      </Root>
    </ThemeProvider>
  );
}
