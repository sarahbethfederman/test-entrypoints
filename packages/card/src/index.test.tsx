import * as React from 'react';
import { mount } from 'enzyme';
import Card, { CardProps } from '.';
import Theme from '@lendi-ui/theme';
import { CardHead, CardBody, MoreContainer, CardTitleIcon } from './index.style';
import Dropdown, { Item } from '@lendi-ui/dropdown';
import { Lock, Info } from '@lendi-ui/icon';
import { deriveSize } from '@lendi-ui/utils';

let wrapper;
const render = (props: CardProps) => {
  wrapper = mount(
    <Theme>
      <Card {...props} />
    </Theme>
  );
};

describe('Card', () => {
  it('renders card without Cancel button', () => {
    render({ title: 'My Testing card' });
    expect(wrapper.find(Card)).toHaveLength(1);
    expect(wrapper.find(CardHead)).toHaveLength(1);
    expect(wrapper.find(CardBody)).toHaveLength(1);
    expect(wrapper.html().indexOf('My Testing card')).not.toEqual(-1);
    expect(wrapper.props().onCancel).toBeUndefined();
  });

  it('renders card with Cancel button', () => {
    render({ title: 'My Testing card', onCancel: () => alert('on cancel clicked') });
    expect(wrapper.find(MoreContainer)).toHaveLength(1);
    expect(wrapper.find(Card).props().onCancel).toBeDefined();
  });

  it('render card with interactive title', () => {
    const items: Item[] = [
      {
        value: '1',
        label: '5 years',
      },
      {
        value: '2',
        label: '10 years',
      },
    ];
    render({ interactiveTitle: <Dropdown items={items} /> });
    expect(wrapper.find(Dropdown)).toBeDefined();
    expect(wrapper.find(Card)).toMatchSnapshot();
  });

  it('render card with icon and cancel button', () => {
    render({
      title: 'My Testing card',
      cardTitleIcon: <Lock color="secondary.500" />,
      onCancel: () => alert('on cancel clicked'),
    });
    expect(wrapper.find(CardTitleIcon)).toHaveLength(1);
    expect(wrapper.find(Card).props().cardTitleIcon).toBeDefined();
    expect(wrapper.find(CardHead)).toHaveStyleRule('font-size', deriveSize(1)); // default size
    expect(wrapper.find(Card)).toMatchSnapshot();
  });

  it('render card with lg headersize', () => {
    render({
      title: 'My Testing card',
      headerSize: 'lg',
      onCancel: () => alert('on cancel clicked'),
    });
    expect(wrapper.find(Card).props().headerSize).toEqual('lg');
    expect(wrapper.find(CardHead)).toHaveStyleRule('font-size', deriveSize(1.5));
    expect(wrapper.find(Card)).toMatchSnapshot();
  });
  it('render card with just more icon', () => {
    render({
      title: 'My Testing card',
      moreIcon: <Info color="secondary.500" />,
      onIconClick: () => alert('Icon clicked!'),
    });
    expect(wrapper.find(Card).props().title).toEqual('My Testing card');
    expect(wrapper.find(Card).props().moreIcon).toEqual(
      expect.objectContaining({
        props: { color: 'secondary.500' },
      })
    );
    expect(wrapper.find(Card).props().onIconClick).toBeInstanceOf(Function);
    expect(wrapper.find(Card)).toMatchSnapshot();
  });
});
