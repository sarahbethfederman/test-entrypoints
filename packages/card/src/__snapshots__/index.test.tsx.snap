// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Footer renders a footer aligned to the center 1`] = `
.c0 {
  padding-top: calc(0.5 * var(--lendi-ui-size));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<Footer
  align="center"
>
  <styled.div
    align="center"
  >
    <StyledComponent
      align="center"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
    padding-top: ",
              "calc(0.5 * var(--lendi-ui-size))",
              ";
  ",
              "
  display: flex;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        Some footer
      </div>
    </StyledComponent>
  </styled.div>
</Footer>
`;

exports[`Card Footer renders a footer aligned to the left 1`] = `
.c0 {
  padding-top: calc(0.5 * var(--lendi-ui-size));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<Footer
  align="left"
>
  <styled.div
    align="left"
  >
    <StyledComponent
      align="left"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
    padding-top: ",
              "calc(0.5 * var(--lendi-ui-size))",
              ";
  ",
              "
  display: flex;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        Some footer
      </div>
    </StyledComponent>
  </styled.div>
</Footer>
`;

exports[`Card Footer renders a footer aligned to the right 1`] = `
.c0 {
  padding-top: calc(0.5 * var(--lendi-ui-size));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<Footer
  align="right"
>
  <styled.div
    align="right"
  >
    <StyledComponent
      align="right"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
    padding-top: ",
              "calc(0.5 * var(--lendi-ui-size))",
              ";
  ",
              "
  display: flex;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        Some footer
      </div>
    </StyledComponent>
  </styled.div>
</Footer>
`;
